action/protocol/execution/evm/evm.go

```diff
@@ -418,6 +418,12 @@ func executeInEVM(ctx context.Context, evmParams *Params, stateDB *StateDBAdapte
 		config = vmCfg
 	}
 	chainConfig := getChainConfig(g, blockHeight, evmParams.evmNetworkID)
+	if g.IsRedsea(blockHeight) {
+		// Merge enabled at Redsea height
+		tipHash := protocol.MustGetBlockchainCtx(ctx).Tip.Hash
```

Comment:
TIP.Hash is not of randomness. Here is ETH's design for reference: https://github.com/ethereum/consensus-specs/blob/v1.4.0-beta.3/specs/phase0/beacon-chain.md#randao

Reply:
yes, in Beacon chain validator's BLS signature is used as the contribution to entropy, and its hash is XORed into the RANDAO during the update. The Beacon chain maintains the states, and the randomness is reflected as `MixDigest   common.Hash` in Ethereum's block header.

Currently we don't have these in place nor the corresponding data field in our block header.  We could implement a similar scheme but that would take quite some time.

Reply:
for evm, `prevrandao()` instruction always return a same value (i.e. equals to Random field of BlockCtx) during one block, according to implementation of ethereum. So it's not randomness during one block.